enumerable.rb

```diff
@@ -0,0 +1,163 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  # 1 my_each
+  def my_each
+    to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+  end
+
+  # 2 my_each_with_index
+  def my_each_with_index
+    to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
+
+  # 3 my_select
+  def my_select
+    to_enum(:my_select) unless block_given?
+
+    my_arr = []
+    to_a.my_each { |item| my_arr << item if yield item }
+    my_arr
+  end
+
+  # 4 my_all
+  def my_all?(param = nil)
+    if block_given?
+      to_a.my_each { |i| return false if yield(i) == false }
+      return true
+    elsif param.nil?
+      to_a.my_each { |i| return false if i.nil? || i == false }
+    elsif !param.nil? && (param.is_a? Class)
+      to_a.my_each { |i| return false unless [i.class, i.class.superclass].include?(param) }
+    elsif !param.nil? and param.instance_of?(Regexp)
+      to_a.my_each { |i| return false unless param.match(i) }
+    else
+      to_a.my_each { |i| return false if i != param }
+    end
+    true
+  end
+
+  # 5 my_any
+  def my_any?(param = nil)
+    if block_given?
+      to_a.my_each { |i| return true if yield(i) }
+      return false
+    elsif param? nil
+      to_a.my_each { |i| return true if i }
+    elsif !param.nil? && (param.is_a? Class)
+      to_a.my_each { |i| return true if [i.class, i.class.superclass].include?(param) }
+    elsif !param.nil? && param.instance_of?(Regexp)
+      to_a.my_each { |i| return true if param.match(i) }
+    else
+      to_a.my_each { |i| return true if i == param }
+    end
+    false
+  end
```

Comment:
#my_any? when no block or argument is given returns true if at least one of the collection is 
not false or nil
#my_any? when no block or argument is given returns false if at least one of the collection is not true
#my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
#my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
#my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
#my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
#my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
#my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern